### DISCUSSION TITLE
About PrimeNG TableRowExpand

### DISCUSSION THREAD

**Discussion by erhanarslanteknosol** [primefaces/community#681] [PrimeNG]
Hello,
I am trying to use TableRowExpand in our project. Because of our relational database, need to manipulate our data as id based. In this current situation, to reach your JSON data, we need to call getProductsData() then getProductsWithOrdersData() methods respectively. How to manage this situation? How to reach orders via product ? how to relate each array objects 
product: [{ 
                id: '1000',
                code: 'f230fh0g3',
                name: 'Bamboo Watch',
                description: 'Product Description',
                image: 'bamboo-watch.jpg',
                price: 65,
                category: 'Accessories',
                quantity: 24,
                inventoryStatus: 'INSTOCK',
                rating: 5,
                }]
 orders: [orders: [
                    {
                        id: '1000-0',
                        productCode: 'f230fh0g3',
                        date: '2020-09-13',
                        amount: 65,
                        quantity: 1,
                        customer: 'David James',
                        status: 'PENDING'
                    },
                    {
                        id: '1000-1',
                        productCode: 'f230fh0g3',
                        date: '2020-05-14',
                        amount: 130,
                        quantity: 2,
                        customer: 'Leon Rodrigues',
                        status: 'DELIVERED'
                    }]

**Comment by cetincakiroglu** [comment]
Hi,

Could you please explain "How to reach orders via product ?". You don't need to use our data in your app, it's only for demonstration. You should create a service, configure your requests, and manipulate your data if needed.
